### DISCUSSION TITLE
Nova editor extension using vAmiga

### DISCUSSION THREAD

**Discussion by DidierMalenfant** [dirkwhoffmann/vAmiga#812] [Show and tell]
I'm toying with making an extension for [nova](nova.app) which would be similar to bartman's [vscode-amiga-debug](https://github.com/BartmanAbyss/vscode-amiga-debug) extension but macOS native.

I have a `vbcc/vasm` toolchain already up and running in Nova and now I'm onto building an ADF image and running it on an emulator. I'd love to include and use vAmiga because the codebase is really clean compared FS-UAE, and it's very macOS friendly.

There are a few things that will need to be added so I'm going to use this thread as a way to make sure I don't wander in the wrong direction with my contribution to the project.

I was thinking that #809 would be a good one for me to start on if no one else is tackling it. I'll post on that issue to seek guidance on how to get started.

**Comment by dirkwhoffmann** [comment]
Your project looks very promising to me. Will your plugin communicate with the emulator via a GDB-like protocol? There is already some code in vAmiga for that (GDB server). It had been implemented a while ago to integrate vAmiga into prb28's vscode plugin (see #618), but this project was never completed.

**Reply by DidierMalenfant** [reply]
Haven't started on the debugging part yet, just focusing on the build and runtime toolchain.

Nova uses the [Debug Adapter Protocol](https://microsoft.github.io/debug-adapter-protocol/). I haven't looked at how close this is to GDB's but I will at some point...
